.github/workflows/build-test-ship.yaml

```diff
@@ -0,0 +1,145 @@
+name: Build, Test, and Ship
+permissions:
+  contents: read
+on:
+  workflow_call:
+    inputs:
+      services:
+        description: >
+          A JSON array containing the names of all services in the repo: the
+          eponymous microservice itself, any associated Kafka consumers, etc.
```

Comment:
This only takes the service names themselves and assumes the corresponding `Dockerfile`s are in `./cmd/<service>/Dockerfile`. At present, this is not universally true. For instance, the `lead` service is at the top level of the repo. @geoffmore's `entity` PR (https://github.com/nicheinc/entity/pull/221/files) instead takes a JSON object containing both the services and their build paths. I think we can simplify this by standardizing our backend repo layouts at the same time we switch the workflow files, so all repos place executables under `cmd`.